### PR TITLE
HOC for forms

### PR DESCRIPTION
**Pull request by chrisnojima**
This creates  HOC to wrap connected components to avoid costly calls through the redux store. This should be a workaround to slow typing.

I implemented this on 3 examples w/ different use cases:

1. Chat: You update the value and make a service call. Debounced to every 500ms
1. Profile: Edit bio/location/name. Used to keep this state locally in the connected component but now it just lives in the HOC. Uses the HOC helper to get access to its internal state values onSubmit()
1. Profile add proof-Enter username: Enter in a value, Debounced check just validates the input looks ok (btc for example). Passes a submit which does validation before the submit (not debounced) so we don't violate any checking

### CODE REVIEW

File: shared/nav.android.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ import NoTab from './no-tab'
 import ProfileContainer from './profile/container'
 import Push from './push/push.native'
 import React, {Component} from 'react'
-import Search from './search'
+import Search from './search/container'
```
Comment (chrisnojima) [inline shared/nav.android.js]: fix naming structure of search

File: shared/profile/edit-profile/index.js [hunk 2]
```diff
@@ -1,22 +1,15 @@
...
 import Render from './render'
+import {HOCForm} from '../../common-adapters'
 import {connect} from 'react-redux'
 import {editProfile} from '../../actions/profile'
 import {maxProfileBioChars} from '../../constants/profile'
 import {navigateUp} from '../../actions/router'
 
 import type {Props} from './render'
 
-type State = {
```
Comment (chrisnojima) [inline shared/profile/edit-profile/index.js]: Don't need state here anymore!

File: shared/profile/prove-enter-username.js.flow [hunk 3]
```diff
@@ -11,7 +11,7 @@ export type Props = {
   errorCode: ?number,
   canContinue?: boolean,
   onUsernameChange?: (username: string) => void,
-  onContinue: (username: string) => void,
```
Comment (chrisnojima) [inline shared/profile/prove-enter-username.js.flow]: This was just incorrect, never used the value

File: shared/search/container.js [hunk 4]
```diff
@@ -1,18 +1,19 @@
 // @flow
 import React, {Component} from 'react'
-import {isMobile} from '../constants/platform'
```
Comment (chrisnojima) [inline shared/search/container.js]: just me sorting this, sorry mgood :(

**Comment by chrisnojima** [comment]
Had to change how search works so its simpler. If you change platforms it'll auto search, we don't have to do it manually in the render component

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

**Comment by keybase-ci-visdiff** [comment]
The commits 9c11b3d689c3...44810408f647 introduce visual changes on linux. <details><summary>:mag_right: 13 changed</summary>

 * Changed: **New_Proof_Enter_Username-Zcash** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Zcash-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Zcash-44810408f647.png)  
   New_Proof_Enter_Username-Zcash comparison
 * Changed: **New_Proof_Enter_Username-Website** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Website-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Website-44810408f647.png)  
   New_Proof_Enter_Username-Website comparison
 * Changed: **New_Proof_Enter_Username-Twitter_with_Error** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter_with_Error-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter_with_Error-44810408f647.png)  
   New_Proof_Enter_Username-Twitter_with_Error comparison
 * Changed: **New_Proof_Enter_Username-Twitter** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter-44810408f647.png)  
   New_Proof_Enter_Username-Twitter comparison
 * Changed: **New_Proof_Enter_Username-Reddit** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Reddit-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Reddit-44810408f647.png)  
   New_Proof_Enter_Username-Reddit comparison
 * Changed: **New_Proof_Enter_Username-Hacker_News** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Hacker_News-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Hacker_News-44810408f647.png)  
   New_Proof_Enter_Username-Hacker_News comparison
 * Changed: **New_Proof_Enter_Username-Git_Hub** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Git_Hub-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Git_Hub-44810408f647.png)  
   New_Proof_Enter_Username-Git_Hub comparison
 * Changed: **New_Proof_Enter_Username-Facebook** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Facebook-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Facebook-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Facebook.png)
 * Changed: **New_Proof_Enter_Username-DNS** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-DNS-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-DNS-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-DNS.png)
 * Changed: **New_Proof_Enter_Username-Coinbase_with_Error** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase_with_Error-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase_with_Error-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase_with_Error.png)
 * Changed: **New_Proof_Enter_Username-Coinbase** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase.png)
 * Changed: **New_Proof_Enter_Username-Bitcoin_Disabled** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin_Disabled-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin_Disabled-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin_Disabled.png)
 * Changed: **New_Proof_Enter_Username-Bitcoin** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479413237278-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin.png)
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 9c11b3d689c3...44810408f647 introduce visual changes on linux. <details><summary>:mag_right: 13 changed</summary>

 * Changed: **New_Proof_Enter_Username-Zcash** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Zcash-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Zcash-44810408f647.png)  
   New_Proof_Enter_Username-Zcash comparison
 * Changed: **New_Proof_Enter_Username-Website** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Website-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Website-44810408f647.png)  
   New_Proof_Enter_Username-Website comparison
 * Changed: **New_Proof_Enter_Username-Twitter_with_Error** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter_with_Error-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter_with_Error-44810408f647.png)  
   New_Proof_Enter_Username-Twitter_with_Error comparison
 * Changed: **New_Proof_Enter_Username-Twitter** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter-44810408f647.png)  
   New_Proof_Enter_Username-Twitter comparison
 * Changed: **New_Proof_Enter_Username-Reddit** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Reddit-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Reddit-44810408f647.png)  
   New_Proof_Enter_Username-Reddit comparison
 * Changed: **New_Proof_Enter_Username-Hacker_News** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Hacker_News-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Hacker_News-44810408f647.png)  
   New_Proof_Enter_Username-Hacker_News comparison
 * Changed: **New_Proof_Enter_Username-Git_Hub** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Git_Hub-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Git_Hub-44810408f647.png)  
   New_Proof_Enter_Username-Git_Hub comparison
 * Changed: **New_Proof_Enter_Username-Facebook** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Facebook-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Facebook-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Facebook.png)
 * Changed: **New_Proof_Enter_Username-DNS** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-DNS-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-DNS-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-DNS.png)
 * Changed: **New_Proof_Enter_Username-Coinbase_with_Error** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase_with_Error-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase_with_Error-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase_with_Error.png)
 * Changed: **New_Proof_Enter_Username-Coinbase** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase.png)
 * Changed: **New_Proof_Enter_Username-Bitcoin_Disabled** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin_Disabled-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin_Disabled-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin_Disabled.png)
 * Changed: **New_Proof_Enter_Username-Bitcoin** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479414962952-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin.png)
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 9c11b3d689c3...44810408f647 introduce visual changes on linux. <details><summary>:mag_right: 13 changed</summary>

 * Changed: **New_Proof_Enter_Username-Zcash** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Zcash-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Zcash-44810408f647.png)  
   New_Proof_Enter_Username-Zcash comparison
 * Changed: **New_Proof_Enter_Username-Website** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Website-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Website-44810408f647.png)  
   New_Proof_Enter_Username-Website comparison
 * Changed: **New_Proof_Enter_Username-Twitter_with_Error** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter_with_Error-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter_with_Error-44810408f647.png)  
   New_Proof_Enter_Username-Twitter_with_Error comparison
 * Changed: **New_Proof_Enter_Username-Twitter** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter-44810408f647.png)  
   New_Proof_Enter_Username-Twitter comparison
 * Changed: **New_Proof_Enter_Username-Reddit** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Reddit-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Reddit-44810408f647.png)  
   New_Proof_Enter_Username-Reddit comparison
 * Changed: **New_Proof_Enter_Username-Hacker_News** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Hacker_News-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Hacker_News-44810408f647.png)  
   New_Proof_Enter_Username-Hacker_News comparison
 * Changed: **New_Proof_Enter_Username-Git_Hub** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Git_Hub-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Git_Hub-44810408f647.png)  
   New_Proof_Enter_Username-Git_Hub comparison
 * Changed: **New_Proof_Enter_Username-Facebook** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Facebook-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Facebook-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Facebook.png)
 * Changed: **New_Proof_Enter_Username-DNS** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-DNS-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-DNS-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-DNS.png)
 * Changed: **New_Proof_Enter_Username-Coinbase_with_Error** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase_with_Error-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase_with_Error-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase_with_Error.png)
 * Changed: **New_Proof_Enter_Username-Coinbase** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase.png)
 * Changed: **New_Proof_Enter_Username-Bitcoin_Disabled** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin_Disabled-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin_Disabled-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin_Disabled.png)
 * Changed: **New_Proof_Enter_Username-Bitcoin** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479418027611-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin.png)
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 9c11b3d689c3...44810408f647 introduce visual changes on linux. <details><summary>:mag_right: 13 changed</summary>

 * Changed: **New_Proof_Enter_Username-Zcash** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Zcash-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Zcash-44810408f647.png)  
   New_Proof_Enter_Username-Zcash comparison
 * Changed: **New_Proof_Enter_Username-Website** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Website-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Website-44810408f647.png)  
   New_Proof_Enter_Username-Website comparison
 * Changed: **New_Proof_Enter_Username-Twitter_with_Error** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter_with_Error-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter_with_Error-44810408f647.png)  
   New_Proof_Enter_Username-Twitter_with_Error comparison
 * Changed: **New_Proof_Enter_Username-Twitter** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Twitter-44810408f647.png)  
   New_Proof_Enter_Username-Twitter comparison
 * Changed: **New_Proof_Enter_Username-Reddit** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Reddit-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Reddit-44810408f647.png)  
   New_Proof_Enter_Username-Reddit comparison
 * Changed: **New_Proof_Enter_Username-Hacker_News** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Hacker_News-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Hacker_News-44810408f647.png)  
   New_Proof_Enter_Username-Hacker_News comparison
 * Changed: **New_Proof_Enter_Username-Git_Hub** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Git_Hub-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Git_Hub-44810408f647.png)  
   New_Proof_Enter_Username-Git_Hub comparison
 * Changed: **New_Proof_Enter_Username-Facebook** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Facebook-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Facebook-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Facebook.png)
 * Changed: **New_Proof_Enter_Username-DNS** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-DNS-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-DNS-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-DNS.png)
 * Changed: **New_Proof_Enter_Username-Coinbase_with_Error** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase_with_Error-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase_with_Error-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase_with_Error.png)
 * Changed: **New_Proof_Enter_Username-Coinbase** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Coinbase.png)
 * Changed: **New_Proof_Enter_Username-Bitcoin_Disabled** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin_Disabled-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin_Disabled-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin_Disabled.png)
 * Changed: **New_Proof_Enter_Username-Bitcoin** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin-9c11b3d689c3.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin-44810408f647.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479420027204-9c11b3d689c3-44810408f647-linux/New_Proof_Enter_Username-Bitcoin.png)
</summary>

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by MarcoPolo** [comment]
I think this is a little too dynamic. It's nice because it's a drop in, but I think in the future this is more complicated than what we gain by this drop in solution. Will review more in a bit, but maybe a better solution is something similar to what @chromakode did with route state?

**Comment by chrisnojima** [comment]
needs more discussion

**Pull request closed without merging**